lib/engine.js

```diff
@@ -150,48 +136,84 @@ function gameWin() {
 
 //click events, need somewhere to live.
 function getPosition(event) {
-  var x = event.x;
-  var y = event.y;
-
-  x -= canvas.offsetLeft;
-  y -= canvas.offsetTop;
-
-  return { x: x, y: y };
+  return { x: event.x - canvas.offsetLeft, y: event.y - canvas.offsetTop };
 }
 
 function selectTile(event) {
   let clickPosition = getPosition(event);
+  lastTile = currentTile;
   currentTile = board.getClickedTile(clickPosition.x, clickPosition.y);
+
+  if (tileIsABuildTile(currentTile) && tileIsAHighlightedBuildTile(lastTile)) {
+    lastTile.type = 'buildTile';
+    currentTile.type = 'buildTileHighlighted';
+    $('#tower-prompt').show();
```

Comment:
There's a lot of repetition here - so I'd recommend breaking out the prompt show and hide into its own function